File: e2e/maestro/tests/task-info/task-info-accessibility.test.ts
import { AllureEpic } from '@/data/AllureEpic';
import { maestroTestCase } from '@/runner/maestroTestCase';

export default maestroTestCase(import.meta.url, {
  name: 'should have no accessibility violations on task info form',
  epic: AllureEpic.ACCESSIBILITY,
  feature: 'View task info',
  tms: '102',
  suite: 'accessibility',
  setupMocks: async (support, context) => {
    const response = support.mock.api.fromResponse(context.createTaskResponse());

    await support.reset();
    await support.mock.api.getTasks(response);
    await support.mock.api.getTasks(response);
    await support.mock.api.getTask(response);
    await support.mock.api.getIsValid(true);

    return context.toMaestroEnv();
  },
});

File: demo-ios/DemoUITests/Test/TaskInfo/TaskInfoAccessibilityTest.swift
import XCTest

final class TaskInfoAccessibilityTest: AccessibilityTestBase, AllureAnnotated {
    static var allureEpic: String { AllureEpic.accessibility }
    static var allureFeature: String { "View task info" }
    static var allureTmsLinks: [String] { ["102"] }

    private var context: TaskTestContext!
    private var response: TaskResponse!

    override func stubApis() throws {
        context = TaskTestContext()
        response = context.createTaskResponse()

        try support.mock.api()
            .getTasks(response)
            .getTask(response)
            .getIsValid(true)
    }

    func testShouldHaveNoAccessibilityViolationsOnTaskInfoForm() {
        // Given
        steps.navigation.openMainPage()
        steps.tasks.openTaskInfoForm(response.title)

        // When
        steps.accessibility.analyze(requiredIdentifiers: [
            "description",
            "detail-description-label",
            "detail-validated-label",
            "valid",
        ])

        // Then
        validate.accessibility.hasNoViolations()
    }
}
